### ISSUE TITLE
Cannot import from .ts in a Custom Extension

### ISSUE THREAD

**Issue by kdheepak** [bokeh/bokeh#5916]
The compiler throws an error saying it is not able to find the file. It looks like only importing from .coffee files are supported right now?

The workaround is to import from a .coffee file that has already imported from a .ts file. That appears to be working, although I'm not sure why.

**Comment by bryevdv** [comment]
ping @mattpap

**Comment by kdheepak** [comment]
Oops. It turns out you cannot import from a .coffee file that has already imported from a .ts file. I'd be curious to hear a workaround for this.

**Comment by mattpap** [comment]
@kdheepak, can you provide a minimal example of this issue?

**Comment by kdheepak** [comment]
Sure. The following is an example where the compiler.py file throws a RunTimeError.

``` python
import bokeh

from bokeh.io import output_notebook, show
from bokeh.layouts import gridplot
from bokeh.models import ColumnDataSource, HoverTool
from bokeh.plotting import figure

output_notebook()

from bokeh.core.properties import Instance
from bokeh.io import output_file, show
from bokeh.models import ColumnDataSource, HoverTool, CustomJS
from bokeh.plotting import figure

JS_CODE = """
import {HoverToolView, HoverTool} from "models/tools/inspectors/hover_tool"
import {values, isEmpty} from "core/util/object"

export class CustomHoverToolView extends HoverToolView
  _move: (e) ->
    super(e)
    console.log(values, isEmpty)

export class CustomHoverTool extends HoverTool
  default_view: CustomHoverToolView
  type: "CustomHoverTool"
  tool_name: "CustomHover"
  icon: "bk-tool-icon-hover"
"""

class CustomHoverTool(HoverTool):
    __implementation__ = JS_CODE

x = list(range(0, 20))
y0 = [abs(xx) for xx in x]
y1 = [xx**2 for xx in x]

# create a column data source for the plots to share
source = ColumnDataSource(data=dict(x=x, y0=y0, y1=y1))

TOOLS = "pan,box_zoom,box_select,help"

# create a new plot and add a renderer
left = figure(tools=TOOLS, width=300, height=300, title=None)
left.circle('x', 'y0', source=source, radius=.5)

# create another new plot and add a renderer
right = figure(tools=TOOLS, width=300, height=300, title=None)
right.circle('x', 'y1', source=source)

left.add_tools(CustomHoverTool())
right.add_tools(CustomHoverTool())

p = gridplot([[left, right]])

show(p)
```


```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-1-03c4276801ad> in <module>()
     60 p = gridplot([[left, right]])
     61 
---> 62 show(p)

/Users/$USER/miniconda2/lib/python2.7/site-packages/bokeh/io.pyc in show(obj, browser, new, notebook_handle)
    298     if obj not in _state.document.roots:
    299         _state.document.add_root(obj)
--> 300     return _show_with_state(obj, _state, browser, new, notebook_handle=notebook_handle)
    301 
    302 

/Users/$USER/miniconda2/lib/python2.7/site-packages/bokeh/io.pyc in _show_with_state(obj, state, browser, new, notebook_handle)
    308 
    309     if state.notebook:
--> 310         comms_handle = _show_notebook_with_state(obj, state, notebook_handle)
    311         shown = True
    312 

/Users/$USER/miniconda2/lib/python2.7/site-packages/bokeh/io.pyc in _show_notebook_with_state(obj, state, notebook_handle)
    331     else:
    332         comms_target = make_id() if notebook_handle else None
--> 333         publish_display_data({'text/html': notebook_div(obj, comms_target)})
    334         if comms_target:
    335             handle = _CommsHandle(get_comms(comms_target), state.document,

/Users/$USER/miniconda2/lib/python2.7/site-packages/bokeh/embed.pyc in notebook_div(model, notebook_comms_target)
    260         js_urls = resources.js_files,
    261         css_urls = resources.css_files,
--> 262         js_raw = resources.js_raw + [script],
    263         css_raw = resources.css_raw_str,
    264         elementid = item['elementid']

/Users/$USER/miniconda2/lib/python2.7/site-packages/bokeh/resources.pyc in js_raw(self)
    386             raw.append('Bokeh.set_log_level("%s");' % self.log_level)
    387 
--> 388         custom_models = gen_custom_models_static()
    389         if custom_models is not None:
    390             raw.append(custom_models)

/Users/$USER/miniconda2/lib/python2.7/site-packages/bokeh/util/compiler.pyc in gen_custom_models_static()
    369     for model in custom_models.values():
    370         compiled = custom_impls[model.full_name]
--> 371         deps_map = resolve_deps(compiled.deps, model.path)
    372 
    373         exports.append((model.name, model.module))

/Users/$USER/miniconda2/lib/python2.7/site-packages/bokeh/util/compiler.pyc in resolve_deps(deps, root)
    365         custom_modules = set([ model.module for model in custom_models.values() ])
    366         missing = set(deps) - known_modules - custom_modules
--> 367         return resolve_modules(missing, root)
    368 
    369     for model in custom_models.values():

/Users/$USER/miniconda2/lib/python2.7/site-packages/bokeh/util/compiler.pyc in resolve_modules(to_resolve, root)
    358                     modules.append((sig, code, deps_map))
    359             else:
--> 360                 raise RuntimeError("no such module: %s" % module)
    361 
    362         return resolved

RuntimeError: no such module: core/util/object
```

**Comment by mattpap** [comment]
@kdheepak, what version of bokeh this report refers to? I tried your code in current `master` and it works without errors.

**Comment by kdheepak** [comment]
I'm on 0.12.4. Did you try this on a custom build? Can you try it on the
last stable release just to see if the issue occurs on your computer as
well?

**Comment by mattpap** [comment]
@kdheepak, the problem is that you are using `core/util/object` module, which isn't available in 0.12.4. This module was added in 0.12.5-dev. You have two options, either use `underscore`  module (which was available in 0.12.4) or switch to most recent dev release of bokeh. Perhaps this error message could be improved.

**Comment by kdheepak** [comment]
Ah. Thanks. I'll close this issue then.
